metpy/calc/thermo.py

```diff
@@ -840,3 +842,180 @@ def relative_humidity_from_specific_humidity(specific_humidity, temperature, pre
     return (100 * units.percent *
             mixing_ratio_from_specific_humidity(specific_humidity) /
             saturation_mixing_ratio(pressure, temperature))
+
+
+@exporter.export
+@check_units('[temperature]', '[pressure]', '[temperature]')
+def get_isentropic_pressure(isentlevs, lev, tmp, *args, **kwargs):
+    r"""Perform isentropic analysis with data in isobaric coordinates.
+
+    Parameters
+    ----------
+    lev : array
+        One-dimensional array of pressure levels
+    tmp : array
+        Array of temperature
+    isentlevs : array
+        One-dimensional array of desired theta surfaces
+
+    Returns
+    -------
+    isentpr : tuple
+        Tuple list with pressure at each isentropic level, followed by each additional
+        argument interpolated to isentopic coordinates.
+
+
+    Other Parameters
+    ----------------
+    args : array, optional
+        Any additional variables will be interpolated to each isentropic level.
+
+    tmpk_out : bool, optional
```

Comment:
This was set as tmpk_out rather than temp_out (what it originally was) since the temperature would be output in Kelvin. It's more intuitive to me, but that can be changed to remain consistent too.